### PR TITLE
[DNM] Swift 4

### PR DESCRIPTION
**Pull request by jsom**

### CODE REVIEW

File: TradeItIosTicketSDK2/TradeItLinkedBrokerManager.swift [hunk 1]
```diff
@@ -274,7 +274,7 @@ import PromiseKit
         onFinished: @escaping () -> Void
     ) {
         // Add missing linkedBrokers
-        let localUserIds = self.linkedBrokers.flatMap { $0.linkedLogin.userId }
+        let localUserIds: [String] = self.linkedBrokers.flatMap { $0.linkedLogin.userId }
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItLinkedBrokerManager.swift]: Without defining the type, Swift 4 would interpret this as `flatMap` all the way down including the strings.

File: TradeItIosTicketSDK2/TradeItStepperInputTableViewCell.swift [hunk 2]
```diff
@@ -31,7 +31,7 @@ class TradeItStepperInputTableViewCell: UITableViewCell {
         }
     }
 
-    func dismissKeyboard() {
+    @objc func dismissKeyboard() {
```
Comment (guillaumedebavelaere) [inline TradeItIosTicketSDK2/TradeItStepperInputTableViewCell.swift]: Why do we need `@objc` here ?

File: TradeItIosTicketSDK2/TradeItPortfolioAccountDetailsViewController.swift [hunk 3]
```diff
@@ -67,7 +67,7 @@ class TradeItPortfolioAccountDetailsViewController: TradeItViewController, Trade
                 cacheResult: true,
                 onSuccess: { _ in
                     self.tableViewManager.updateAccount(withAccount: linkedBrokerAccount)
-                    fulfill()
+                    fulfill(())
```
Comment (guillaumedebavelaere) [inline TradeItIosTicketSDK2/TradeItPortfolioAccountDetailsViewController.swift]: What do the extra parenthesis mean ?

File: TradeItIosTicketSDK2/TradeItStepperInputTableViewCell.swift (hunk 2 above)
Comment (jsom) [reply TradeItIosTicketSDK2/TradeItStepperInputTableViewCell.swift]: Functions passed to `#selector` need to be available to Obj-C. In Swift 3 it was inferred, but Swift 4 they are removing that inference.

**Pull request closed without merging**